### ISSUE TITLE
Has anyone used the 2048 uv

### ISSUE THREAD

**Issue by Michaelwhite34** [abdallahdib/NextFace#29]

**Comment by Michaelwhite34** [comment]
The map is good for 512 but higher res suffers OOM and cpu is taking too long, if anyone has a 2048 result, plz share it.

**Comment by abdallahdib** [comment]
u may want to try the gpu backend ?

**Comment by abdallahdib** [comment]
how many samples are u using for optimization ? I never tested with 2048 texture size, u may need to find an alternative (upscaling as post processing to get the desired resolution)

**Comment by Michaelwhite34** [comment]
> how many samples are u using for optimization ? I never tested with 2048 texture size, u may need to find an alternative (upscaling as post processing to get the desired resolution)

Currently I am using Cpu all the way, takes 10-15 hours to get reasonable result.

**Comment by abdallahdib** [comment]
that seems too much, can u tell how many ray tracing samples are u using for optim?

**Comment by Michaelwhite34** [comment]
I am using 100, maybe lower will work.But 8 is to low.

**Comment by Michaelwhite34** [comment]
> that seems too much, can u tell how many ray tracing samples are u using for optim?

8 sample will get around 20-25% average brightness difference in the end,while 100 sample get around 13-14%.

**Comment by abdallahdib** [comment]
so in fact the slow time are caused by these factors in ur case
- texture resolution 2048
- env map 2048
- samples per pxiel (100)
- running on cpu

in fact the default parameters provided by nextFace try to balance between quality and execution time.. u pushed to the max in ur case...  

Also using 100 samples should that much improvment compared to 8.

**Comment by Michaelwhite34** [comment]
Also you can add a script to render with redner using the exported maps, that way we can know if the maps are actually good.

**Comment by abdallahdib** [comment]
well the final output is rendered with redner (saveOutput function in optimizer.py) this function render the face at high number of samples  and save the outpout as an image (render.png) in the given output directory

**Comment by Michaelwhite34** [comment]
> well the final output is rendered with redner (saveOutput function in optimizer.py) this function render the face at high number of samples and save the outpout as an image (render.png) in the given output directory

I am not sure about this, I thought the color information stored in pickle file is not in the format of texture. So rendering these information into texture might somehow break these information, like low resolution environment.

**Comment by abdallahdib** [comment]
all the optimized textures are saved in png format with gamma correction applied to them

**Comment by Michaelwhite34** [comment]
> 

So it means we don't need to gamma correct again to use it ?I understand png is lossless, but I just want to make sure things don't go wrong during the conversion process.

**Comment by abdallahdib** [comment]
yes the function saveImage in Image.py already apply a gamma correction to the image. so u dont need to gamma correct the textures. u can (optionally) modify the code to export to an exr format to get the full range

**Comment by Michaelwhite34** [comment]
> yes the function saveImage in Image.py already apply a gamma correction to the image. so u dont need to gamma correct the textures. u can (optionally) modify the code to export to an exr format to get the full range

I think for now we need to make the final render as close to original image as possible, ideally something like jpeg compression. My result is like 5 to 7 times big as that.
